subsys/dfu/img_util/flash_img.c

```diff
@@ -83,3 +88,82 @@ int flash_img_init(struct flash_img_context *ctx)
 {
 	return flash_img_init_id(ctx, FLASH_AREA_IMAGE_SECONDARY);
 }
+
+#if defined(CONFIG_IMG_ENABLE_IMAGE_CHECK)
+int flash_img_check_sha256(struct flash_img_context *ctx,
+			   const struct flash_img_check *fic)
+{
+	unsigned char hash[TC_SHA256_DIGEST_SIZE];
+	struct tc_sha256_state_struct sha256sum;
+	int to_read;
+	int pos;
+	int rc;
+
+	if (!fic->match) {
```

Comment:
Maybe an assert to ensure fic is not null?

Reply:
This is an internal method and fic was already tested. However, I agreed that this check could be useful in future.
Try cover more in tests.